### PR TITLE
Ginkgo: Added Makefile

### LINKED ISSUE #2089: Ginkgo cleanups
- [ ] Rename readme to `README.md`
- [ ] Add Makefile
  - [ ] Add target to reprovision
  - [ ] Add target to rerun tests while considering local changes

### PR DESCRIPTION
**Pull request by eloycoto**
Created a new Makefile to run ginkgo tests without knowing all the
options.

Rename readme.md to README.md

Fix #2089

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

File: test/Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+provision = true
```
Comment (joestringer) [inline test/Makefile:1]: We can do something like the following to parameterize this:
```
PROVISION?=true
```

Then it's easier to change it on the commandline, I think it should work something like `make run PROVISION=false`.

(referencing https://stackoverflow.com/questions/2826029/passing-additional-variables-from-command-line-to-make)

Comment (eloycoto) [reply test/Makefile:1]: Hi, 

It's working like that:

```
$ git:(GinkgoMakefile) ✗ make run provision=false  
ginkgo --focus " Runtime*" -v -- --cilium.provision=false  
$ git:(GinkgoMakefile) ✗ make run provision=true 
ginkgo --focus " Runtime*" -v -- --cilium.provision=true   
```

Comment (joestringer) [reply test/Makefile:1]: Ah, I didn't realise that Makefiles will not overwrite variables passed like that. OK :)

**Comment by ianvernon** [comment]
Looks good. Can you use these targets for the separate stages in `ginkgo.Jenkinsfile`?

**Review by tgraf — CHANGES_REQUESTED** [review]
Looks good, please merge the commits together.

**Comment by joestringer** [comment]
It'd be good to get a `make all` target as well.

**Review by nebril — APPROVED** [review]
LGTM, please squash

**Review by aanm — APPROVED** [review]

**Review by ianvernon — APPROVED** [review]
LGTM; please create a follow-up GH issue to use these Makefile targets in the Jenkinsfile.

**Comment by ianvernon** [comment]
@eloycoto please rebase / ensure that the build succeeds so we can merge this

**Comment by eloycoto** [comment]
@tgraf  this should be ready-to-merge. Could you validate please?

**Comment by eloycoto** [comment]
test this please

**Comment by ianvernon** [comment]
Merging this as @tgraf 's only concern was having the commits merged together.

**Pull request merged**